plasmoid/contents/ui/ConfigGeneral.qml

```diff
@@ -17,22 +17,76 @@ Item {
     width: childrenRect.width
     height: childrenRect.height
 
+    property alias cfg_locationEntry: locationEntry.checked
+    property alias cfg_woeidEntry: woeidEntry.checked
     property alias cfg_woeid: woeidField.text
     property alias cfg_interval: intervalField.text
     property alias cfg_timeFormat24: timeFormat24Field.checked
     property alias cfg_useWxFonts: useWxFontsField.checked
+    property bool locChecked: plasmoid.configuration.locationEntry
+    property bool woeChecked: plasmoid.configuration.woeidEntry
+
+    function onLocClicked() {
+        if (woeChecked && plasmoid.configuration.locationEntry) { 
+            cfg_woeid = plasmoid.configuration.woeid
+        } else if (!locChecked) {
+                cfg_woeid = ""
+        }
+        woeChecked = false  
+        locChecked = true
+    }
+
+    function onWoeClicked() {
+        if (locChecked && plasmoid.configuration.woeidEntry) { 
+            cfg_woeid = plasmoid.configuration.woeid
+        } else if (!woeChecked) {
+                cfg_woeid = ""
+        }
+        locChecked = false
+        woeChecked = true
+    }
 
     ColumnLayout {
+        GroupBox {
+            title: i18n("Select Location or WOEID entry.")
+            flat: true
+            
+            ColumnLayout {
+                ExclusiveGroup {
+                    id: locationOrWoeidGroup
+                }
+
+                RadioButton {
+                    id: locationEntry 
+                    text: i18n("Enter Location below.")
+                    exclusiveGroup: locationOrWoeidGroup
+                    onClicked: onLocClicked() 
+                }
+
+                RadioButton {
+                    id: woeidEntry 
+                    text: i18n("Enter WOEID below.")
+                    exclusiveGroup: locationOrWoeidGroup
+                    onClicked: onWoeClicked() 
+                }
+            }
+        }
         RowLayout {
             Label {
-                text: i18n("WOEID")
+                text: woeChecked ? i18n("WOEID") : i18n("Location")
```

Comment:
I don't think we need to show label here since the radio button already tells user what it's being entered